content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -867,6 +867,47 @@ is disabled, `.spec.completions` is immutable.
 Use cases for elastic Indexed Jobs include batch workloads which require 
 scaling an indexed Job, such as MPI, Horovord, Ray, and PyTorch training jobs.
 
+### Recreation only when Pod has been fully terminated
+
+{{< feature-state for_k8s_version="v1.28" state="alpha" >}}
+
+For distributed jobs, it is common to have many replicas running in a job.  
```

Comment:
I would suggest to rewrite this paragraph a little keeping in mind the guidance style suggestions: https://kubernetes.io/docs/contribute/style/style-guide/. In particular "use present tense" or "Address the reader as "you"". Maybe something along the lines (but feel free to rephrase also this, just consider as a general direction):
```
By default, the Job controller recreates pods as soon they are either failed or have a deletion timestamp. 

This means, that you may have two running at the same time, both corresponding to the same pod. In that case, one is terminating, while the other is starting up. 

This behavior might be problematic in some situations, including:
- when using a framework which requires only a single pod per index (for example TF, JAX) <- maybe with link to the issue
- when using a cluster autoscaler, which could increase the size of the cluster for only a short while during pod termination.

If this behavior is problematic for your Jobs, you may choose to create replacement pods only when the terminating pod is fully terminal. To do this, set the `.spec.podReplacementPolicy: Failed`.